packages/apps/plugins/plugin-inbox/src/components/Email/Mailbox.tsx

```diff
@@ -49,31 +49,34 @@ const Mailbox = ({ mailbox, options = {} }: MailboxProps) => {
     };
   }, [selected]);
 
-  const messages = [...mailbox.messages]
-    .filter(nonNullable)
+  const messages = [...(mailbox.messages ?? [])]
+    .filter(nonNullable) // TODO(burdon): Why is this necessary?
```

Comment:
iirc @yarolegovich said it shouldn't be but I've found in the past with stacks that sometimes list of references sometimes come back without all of the resolved. I'm not sure if this is a bug or if it's related to deleted objects but we decided to leave the new type this way to force the undefined to be handled because then you won't get unexpected results.

Reply:
sure. it's a strange DX though. @dmaretskyi something to think about in the next API pass. e.g., query option to filter null refs; or make null by default.